packages/cli/src/commands/config/config.ts

```diff
@@ -1,9 +1,12 @@
-/**
- * @flow
- */
-import {type ConfigT} from 'types';
+import {
+  Config,
+  AndroidDependencyConfig,
+  IOSDependencyConfig,
+} from '@react-native-community/cli-types';
 
-function isValidRNDependency(config) {
+function isValidRNDependency(
+  config: AndroidDependencyConfig | IOSDependencyConfig | null,
```

Comment:
Just an idea, but maybe we want to the Dependency in ```@react-native-community/cli-types``` as an interface like:

```typescript
export interface Dependency {
  [key: string]: {
    name: string;
    root: string;
    platforms: {
      android?: AndroidDependencyConfig | null;
      ios?: IOSDependencyConfig | null;
      [key: string]: any;
    };
    assets: string[];
    hooks: {
      prelink?: string;
      postlink?: string;
    };
    params: InquirerPrompt[];
  };
}
```
Don't forgetting to update the Config interface to ```dependencies: Dependacy; ``` and here we can use:

```typescript
import {
  Dependency
} from '@react-native-community/cli-types';

function isValidRNDependency(
  config: Dependency
) {
  return (
    Object.keys(config.platforms).filter(key => Boolean(config.platforms[key]))
      .length !== 0 ||
    (config.hooks && Object.keys(config.hooks).length !== 0) ||
    (config.assets && config.assets.length !== 0) ||
    (config.params && config.params.length !== 0)
  );
}
```

It would solve most of the problems breaking in this file

Reply:
Yea, makes sense to extract this. We used this type in Flow files: https://github.com/react-native-community/cli/blob/fde227d8867e921f6df500c448ab49250f3bcfb6/types/index.js#L172-L175
@Esemesek can you take it over?

Reply:
Done.

Reply:
Nice!